src/Cryptol/Eval/Value.hs

```diff
@@ -122,15 +132,22 @@ data GenValue sym
   | VInteger !(SInteger sym)                   -- ^ @ Integer @ or @ Z n @
   | VRational !(SRational sym)                 -- ^ @ Rational @
   | VFloat !(SFloat sym)
-  | VSeq !Integer !(SeqMap sym (GenValue sym)) -- ^ @ [n]a   @
-                                               --   Invariant: VSeq is never a sequence of bits
+  | VSeqCtor !Integer !(SeqMap sym (GenValue sym)) -- ^ @ [n]a   @
+                                                   --   Invariant: VSeq is never a sequence of bits
```

Comment:
The Haddocks here still refer to `VSeq`, even though the data constructor has been renamed to `VSeqCtor`. We should update the Haddocks to be in sync with the implementation.

Also, it would be worth adding a comment that `VSeqCtor` is intentionally not exported in an effort to preserve the invariant described in the comment, and that all `VSeqCtor` values must be created through a smart constructor such as `mkSeq` or `mkFinSeq`.